state-chain/pallets/cf-swapping/src/tests/dca.rs

```diff
@@ -623,6 +472,138 @@ fn dca_with_fok_fully_executed() {
+				refund_params: None,
+				dca_params: Some(dca_params.clone()),
+				output_address: (*EVM_OUTPUT_ADDRESS).clone(),
+				is_ccm: false,
+			};
+			insert_swaps(&[swap_params]);
+
+			// Check that the swap request was received;
+			assert_has_matching_event!(
+				Test,
+				RuntimeEvent::Swapping(Event::SwapRequested {
+					swap_request_id: SWAP_REQUEST_ID,
+					input_amount: INPUT_AMOUNT,
+					dca_parameters,
+					..
+				}) if dca_parameters == &Some(dca_params.clone())
+			);
+
+			// Check that the first chunk was scheduled
+			assert_has_matching_event!(
+				Test,
+				RuntimeEvent::Swapping(Event::SwapScheduled {
+					swap_request_id: SWAP_REQUEST_ID,
+					swap_id: 1,
+					input_amount,
+					..
+					// All chunks should be 0 amount except the last one
+				}) if *input_amount == ZERO_CHUNK_AMOUNT
+			);
+
+			// Check the DCA state is correct
+			assert_eq!(
+				get_dca_state(SWAP_REQUEST_ID),
+				DcaState {
+					status: DcaStatus::ChunkScheduled(1),
+					// Still the full amount remaining because the first chunk is 0
+					remaining_input_amount: INPUT_AMOUNT,
+					remaining_chunks: NUMBER_OF_CHUNKS - 1,
+					chunk_interval: CHUNK_INTERVAL,
+					accumulated_output_amount: 0,
+				}
+			);
+		})
+		.then_execute_at_block(CHUNK_1_BLOCK, |_| {})
+		.then_execute_with(|_| {
+			assert_has_matching_event!(
+				Test,
+				RuntimeEvent::Swapping(Event::SwapExecuted {
+					swap_request_id: SWAP_REQUEST_ID,
+					swap_id: 1,
+					input_amount,
+					output_amount,
+					..
+					// The first chunk should 0 in and out
+				}) if *input_amount == ZERO_CHUNK_AMOUNT && *output_amount == ZERO_CHUNK_AMOUNT
+			);
+
+			assert_has_matching_event!(
+				Test,
+				RuntimeEvent::Swapping(Event::SwapScheduled {
+					swap_request_id: SWAP_REQUEST_ID,
+					swap_id: 2,
+					input_amount,
+					..
+					// All chunks should be 0 amount except the last one
+				}) if *input_amount == ZERO_CHUNK_AMOUNT
+			);
+
+			assert_eq!(
+				get_dca_state(SWAP_REQUEST_ID),
+				DcaState {
+					status: DcaStatus::ChunkScheduled(2),
+					remaining_input_amount: INPUT_AMOUNT,
+					remaining_chunks: NUMBER_OF_CHUNKS - 2,
+					chunk_interval: CHUNK_INTERVAL,
+					// Should still be 0
+					accumulated_output_amount: 0,
+				}
+			);
+		})
+		.then_execute_at_block(CHUNK_2_BLOCK, |_| {})
+		.then_execute_at_block(CHUNK_3_BLOCK, |_| {})
+		.then_execute_with(|_| {
+			assert_eq!(SwapRequests::<Test>::get(SWAP_REQUEST_ID), None);
+			assert_has_matching_event!(
+				Test,
+				RuntimeEvent::Swapping(Event::SwapExecuted {
+					swap_request_id: SWAP_REQUEST_ID,
+					swap_id: 3,
+					input_amount,
+					output_amount,
+					..
+					// The last chunk should be the full amount
+				}) if *input_amount == INPUT_AMOUNT && *output_amount == OUTPUT_AMOUNT
+			);
+
+			assert_has_matching_event!(
+				Test,
+				RuntimeEvent::Swapping(Event::SwapRequestCompleted {
+					swap_request_id: SWAP_REQUEST_ID
+				}),
+			);
+		});
+}
+
 mod ccm_tests {
```

Comment:
nit: I think it makes more sense for these to live in the same module as the other ccm tests.

Reply:
What's the difference? These tests are also DCA, so they are not exactly out of place here.

Reply:
There are a few other ccm test that are not in the ccm module because i wanted to keep them with other test of the same type. eg, `reject_invalid_ccm_deposit`, `swap_excess_are_confiscated_ccm_via_deposit` ...
So i think its fine to leave them here. The ccm module is for ccm specific behaviour test.

Reply:
My view is basically this:
- ccm can be fok and/or dca. 
- Non-ccm can also be fok and/or dca. 
- CCM is not really a sub-category of dca swaps. 

Therefore a module hierarchy that reflects this would make more sense IMO.

Reply:
I think it would also be incorrect to say that DCA is a subcategory of CCM 🤷. Conceptually, I don't think there is a big difference, either way a test combining them will look out of place. FWIW fill-or-kill tests also include CCM tests already. I think we should take the pragmatic approach and decide where to put a test based on where it can benefit more from feature-specific helpers (which might very well be the CCM module in this case, I don't know, but I don't think the difference would be significant).